win32ss/gdi/ntgdi/freetype.c

```diff
@@ -1397,8 +1397,23 @@ IntGdiLoadFontsFromMemory(PGDI_LOAD_FONT pLoadFont)
         {
             if (FT_IS_SFNT(Face))
             {
-                Status = DuplicateUnicodeString(&Entry->FaceName, pValueName);
-                // TODO: Check return value
+                // L"Name StyleName\0"
+                Length = NameLength + sizeof(L' ') + Entry->StyleName.Length + sizeof(UNICODE_NULL);
+                pszBuffer = ExAllocatePoolWithTag(PagedPool, Length, TAG_USTR);
+                if (pszBuffer)
+                {
+                    RtlInitEmptyUnicodeString(pValueName, pszBuffer, (USHORT)Length);
```

Comment:
Remove this cast please. Casting is only warranted if you've actually checked for overflow and made sure it's impossible. Otherwise you're silencing MSVC's warning but not fixing what it's warning about.

I'd recommend actually fixing the overflow possibility here: simply bail out above if `Length > UNICODE_STRING_MAX_BYTES` (or if this string is purely for display, perhaps use the minimum of the two). After you do that, the cast would be the right thing to do.

Reply:
Removed these casts.